crates/bevy_render/src/render_graph/node_slot.rs

```diff
@@ -49,9 +49,14 @@ impl ResourceSlots {
         slot.resource = Some(resource);
     }
 
-    pub fn get(&self, label: impl Into<SlotLabel>) -> Option<RenderResourceId> {
+    pub fn get(&self, label: impl Into<SlotLabel>) -> &Option<RenderResourceId> {
         let slot = self.get_slot(label).unwrap();
-        slot.resource.clone()
+        &slot.resource
+    }
```

Comment:
This change isn't technically needed, but it was really surprising to me to be passed an owned copy, rather than a reference. It also saves on a few `clone`s elsewhere.

Reply:
RenderResourceId should almost certainly be a Copy type in its current form (due to its size and the fact that things like TextureId are Copy). Passing a reference here doesn't really provide anything and (according to my understanding) often ends up being slower.

There is a world where we move to RAII for render resources. Then I think using references makes sense here to avoid the cost of incrementing the ref counter. But until then I think my vote is still "get() returns an owned type and set() is used to set values".

Reply:
And we should probably derive Copy on RenderResourceId.

Reply:
Using references here also affects the ergonomics of the situation (requires `*resource` to use the returned value)

Reply:
True enough. Copy is better, it was the clone that pushed it over the top for me, as I prefer those being explicit. I'll make it Copy and change it back.

Reply:
Done.